src/app/app.component.html

```diff
@@ -42,14 +42,15 @@ <h2>ngx-uploader</h2>
                   </div>
                   <div class="progress-content">
                     <div class="progress">
-                      <span class="bar" [style.width]="f?.progress?.data?.percentage + '%'" [class.is-done]="f?.progress?.data?.percentage === 100"></span>
+                      <span class="bar" [style.width]="f?.progress?.data?.percentage + '%'" [class.is-done]="f?.progress?.data?.percentage === 100" [class.is-failed]="f.progress?.status === 4"></span>
                     </div>
                   </div>
                   <div class="progress-text-content">
-                    <span class="progress-text" [class.is-done]="f?.progress?.data?.percentage === 100">
+                    <span class="progress-text" [class.is-done]="f?.progress?.data?.percentage === 100" [class.is-failed]="f.progress?.status === 4"></span>
                       <span>{{ f.progress?.data?.percentage }}% </span>
                       <span *ngIf="f.progress?.data?.percentage !== 100">Uploading...</span>
-                      <span *ngIf="f.progress?.data?.percentage === 100">Done</span>
+                      <span *ngIf="f.progress?.status !== 4 && f.progress?.data?.percentage === 100">Done</span>
+                      <span *ngIf="f.progress?.status === 4">Failed<span *ngIf="!!f.response?.detail"> [{{ f.response?.detail }}]</span></span>
```

Comment:
you are closing <span/> here twice, please fix.

Reply:
There is an inner span...

Reply:
I try to run demo locally and got template compile errors, please check.

Reply:
Fixed... strange that I missed that. Thanks!